config/configunmarshaler/defaultunmarshaler.go

```diff
@@ -65,12 +65,18 @@ const (
 	pipelinesKeyName = "pipelines"
 )
 
+type ConfigSourceSettings struct {
+	ConfigSources []map[string]map[string]interface{} `mapstructure:"config_sources"`
+	MergeConfigs  []string                            `mapstructure:"merge_configs"`
+}
```

Comment:
Did we not discussed that this is a separate File?

Reply:
Yes, we did. This is an slightly different approach to avoid adding a different file and new command line option, see PR description.

Reply:
To clarify, it still can be a separate file or it can be the same file, both ways are valid and the user can decide how they want to use it.

Reply:
What does it happen if the source changes the config sources?